src/Orleans.Transactions/DistributedTM/TransactionOverloadDetector.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Orleans.Transactions
+{
+    public interface ITransactionOverloadDetector
```

Comment:
Not fond of OverloadDetector name, but consistency is more important than whether I like the name.

In this case the interface, implementation, and options are inconsistently named.  The TransactionOverloadDetector is a specialization of the interface, and name needs reflect the specialization's behavior (which it does not) and the options are associated with the specialization and should reflect it's naming, which it does not.

Reply:
that depends on the design. I was intending to add more algorithm to TransactionOverloadDetector. So it will include more logic/algorithm in the future. Then in this case, calling it TransactionOverloadDetector  is consistent. But if you want to have multiple overload detector, every of which just focus on one type of load shedding , then maybe we should rename it... Not sure we want to go that route, we can discuss